### PR TITLE
Update instructions for updating dev environment.

### LINKED ISSUE #7158: Improve docs on the development mode installation
### Description:

Sections 2.1, 2.2 in https://scikit-image.org/docs/stable/user_guide/install.html#id4 .

### Way to reproduce:

_No response_

### Version information:

_No response_

### PR DESCRIPTION
**Pull request by mkcor**
## Description

Closes #7158.

@soupault I have only updated section 2.2 https://scikit-image.org/docs/stable/user_guide/install.html#updating-the-installation -- what's the issue you've encountered when following instructions in section 1.2?

## Checklist

- A descriptive but concise pull request title
- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [Unit tests](https://scikit-image.org/docs/dev/development/contribute.html#testing)
- A gallery example in `./doc/examples` for new features
- [Contribution guide](https://scikit-image.org/docs/dev/development/contribute.html) is followed

## Release note

Summarize the introduced changes in the code block below in one or a few sentences. The
summary will be included in the next release notes automatically:

```release-note
...
```

### CODE REVIEW

File: INSTALL.rst [hunk 1]
```diff
@@ -302,16 +302,28 @@ Test your installation for correct behavior using:
...
-    git checkout main
-    git pull upstream main
-    # Update the installation
-    pip install -e . -vv
+   git checkout main
+   git pull upstream main
+
+And you possibly want to create a feature branch from there (or check out an
+existing branch where you are testing changes). As you work on this branch,
+you can re-build scikit-image with the following command:
```
Comment (stefanv) [inline INSTALL.rst:315]:
```suggestion
And you likely want to create a feature branch from there.
As you work on this branch, you can re-build scikit-image using:
```

File: INSTALL.rst [hunk 2]
```diff
@@ -302,16 +302,28 @@ Test your installation for correct behavior using:
...
+
+.. code-block:: sh
+
+   spin build
+
+If you need to clean the build directory before building scikit-image (e.g.,
```
Comment (stefanv) [inline INSTALL.rst:321]: How about something:

"Mostly, repeated, incremental builds work just fine, but if you notice strange build problems,
rebuild from scratch using:"

Comment (mkcor) [reply INSTALL.rst:320]: Thanks, @stefanv. Editing your suggestion slightly, how about:

"Repeated, incremental builds usually work just fine, but if you notice build problems,
rebuild from scratch using:"

(I find that the word 'strange' creates some unnecessary mystification...)

Comment (stefanv) [reply INSTALL.rst:320]: :ghost:

**Review by stefanv — APPROVED** [review]

**Comment by stefanv** [comment]
Approved pending the change discussed.

**Review by soupault — COMMENTED** [review]
@mkcor this is great, thanks!

About 2.1: In general, I would expect to have a fully working dev installation of scikit-image after following the steps from Section 2. Accordingly, I suggest to add the env variable business at the end of 2.1.1 and 2.1.2 (or the beginning of 2.2):
```python
# Try the new version in IPython
spin ipython
# Copy PYTHONPATH found by spin (usually, `${PWD}/build-install/...`)
# Set for the current session
export PYTHONPATH=${PYTHONPATH}:%enter the path here%
```

Additionally, this line https://github.com/scikit-image/scikit-image/blob/main/meson.md?plain=1#L54 does not generalize to all the users. My proposed change is the same as above:
```python
spin ipython
# Copy PYTHONPATH found by spin (usually, `${PWD}/build-install/...`)
# Set for the current session
export PYTHONPATH=${PYTHONPATH}:%enter the path here%
```

**Review by lagru — APPROVED** [review]
LGTM, even without the pending suggestion in https://github.com/scikit-image/scikit-image/pull/7160#pullrequestreview-1648923672.

**Comment by soupault** [comment]
@lagru got it, thanks! I think I would still prefer having a bit more clarity in this part (2.1.1/2.1.2), e.g.:
```python
...
# The built version is now located under `${PWD}/build-install/...`
# Test your installation
spin test
...
```

**Comment by stefanv** [comment]
Yes, no need to set variables by hand. Also see `spin run` for executing arbitrary commands with PYTHONPATH set.

**Comment by mkcor** [comment]
@soupault is 33e5bf671da1d414acd263a7e254284ff6a0e07e good enough?

I just noticed that the `venv` instructions end with:
```sh
# Try the new version in IPython
spin ipython
```
whereas the `conda` instructions end with:
```sh
# Try the new version
spin python
```
Is there a reason for this difference?

**Comment by lagru** [comment]
> Is there a reason for this difference?

For `venv` ipython is installed via `requirements.txt` > `developer.txt`. For `conda` it's not and as far as I can see therefore missing...

**Comment by lagru** [comment]
I think ipywidgets is pulling it in as an implicit dependency. Probably should make it explicit and add it to our developer dependencies?

**Comment by soupault** [comment]
>is https://github.com/scikit-image/scikit-image/commit/33e5bf671da1d414acd263a7e254284ff6a0e07e good enough?

Happy with that! 🙂 I think https://github.com/scikit-image/scikit-image/blob/main/meson.md?plain=1#L54 should be updated with the same path.

**Comment by mkcor** [comment]
> I think ipywidgets is pulling it in as an implicit dependency.

I remember noticing something like this in the past and though, of, that's convenient... :wink: 

> Probably should make it explicit and add it to our developer dependencies?

So far I've thought 'why bother?' but I'm definitely not against this (explicit is better than implicit?). Should this be another PR? Otherwise I can add it here by tomorrow.

**Comment by mkcor** [comment]
> Otherwise I can add it here by tomorrow.

Sorry, Saturday I was sick. Done b20f24ddd0f0319eef51408b05694de975a3786f.

**Comment by mkcor** [comment]
> > is [33e5bf6](https://github.com/scikit-image/scikit-image/commit/33e5bf671da1d414acd263a7e254284ff6a0e07e) good enough?
> 
> Happy with that! 🙂

:+1: 

> I think https://github.com/scikit-image/scikit-image/blob/main/meson.md?plain=1#L54 should be updated with the same path.

It's never been clear to me whom this `meson.md` file targets... And I believe these `meson` [instructions](https://github.com/scikit-image/scikit-image/blob/3eb9e816aeaf0085b4b77255917b884318a6667c/meson.md?plain=1#L37) are somewhat unrelated to what we've been discussing about `spin`(?). Certainly, I don't have any path like
https://github.com/scikit-image/scikit-image/blob/3eb9e816aeaf0085b4b77255917b884318a6667c/meson.md?plain=1#L54
containing the `skimage` package. I have just `${PWD}/build/` and `${PWD}/build-install/usr/lib/python3.9/site-packages`. So this line probably needs fixing indeed. Can we address this in another PR, because I'm not very meson-knowledgeable?

**Comment by soupault** [comment]
@mkcor thank you for working on this! We surely can address the rest separately. 🙂

**Comment by stefanv** [comment]
The separate meson instructions can go away now; they were there to help those of us working on the transition, but `spin` has now become a much easier UI to doing the same.

**Pull request merged**